finite_difference/src/subdomain_mod.f90

```diff
@@ -162,7 +162,8 @@ function decompose(domainx, domainy,             &
     end if
 
     decomp%ndomains = ndom
-    allocate(decomp%subdomains(ndom), Stat=ierr)
+    ! Ranks start with 0, so allocate the subdomains starting with 0
```

Comment:
Is there a reason for this change? In my other work (#7) I've been working on the principle that, since this is Fortran, we'll add one to the MPI rank. I can go either way :-)

Reply:
Oh yes, so I did :-( I think, in my defence, that that is within the decomposition code. My plan is that publicly it will be 1-indexed. (I'll have to check what I've done and then you can review it :-) )